controllers/pvc-mutator/encryption/keys/manager.go

```diff
@@ -0,0 +1,174 @@
+package keys
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/storageos/api-manager/internal/pkg/crypto"
+)
+
+var (
+	// ErrNoKeyInSecret is returned is a secret was expected to contain an
+	// encryption key, but didn't.
+	ErrNoKeyInSecret = errors.New("secret does not contain encryption key")
+)
+
+// KeyManager generates, stores and removes encryption keys.
+type KeyManager struct {
+	client client.Client
+}
+
+// New - creates a new DefaultManager.
+func New(client client.Client) *KeyManager {
+	return &KeyManager{client: client}
+}
+
+// Ensure that a secret exists at volKeyRef, creating it with valid keys if
+// needed.  If a secret already exists, it does not verify validity.
+func (m *KeyManager) Ensure(ctx context.Context, nsKeyRef client.ObjectKey, volKeyRef client.ObjectKey) error {
+	// Return immediately if the volume key already exists, or an error if
+	// another error (e.g. permission denied).
+	err := m.client.Get(ctx, volKeyRef, &corev1.Secret{})
+	if err == nil {
+		return nil
+	}
+	if !apierrors.IsNotFound(err) {
```

Comment:
This if statement technically belongs to the previous. In that case i ususally prefer `else if`.

Reply:
I'm not sure I agree with this one.  Where an error handling case results in a return, then there's no need to add an `else`.  I find this easier to read, especially when not adding a space between lines to indicate that it's a related block.

Reply:
I can agree. At the concrete situation both ifs are working on the same data. The shortest/lower_scope version looks this:
```
if err := m.client.Get(ctx, volKeyRef, &corev1.Secret{}); err == nil {
    return nil
} else if !apierrors.IsNotFound(err) {
    return err
}
```
I can Go further :D 
```
if err := m.client.Get(ctx, volKeyRef, &corev1.Secret{}); err == nil || !apierrors.IsNotFound(err) {
    return err
}
```

I mark this resolved, because your view also makes sense.

Reply:
oh, I like your last suggestion!  Changed.

Reply:
I found an other place: https://github.com/storageos/api-manager/pull/45/files#diff-de1ea10f4b2cc388d0a7f3bdbc4fd1dfedacb0d1353361dcaf7e193426acff18R54